compiler/src/dotty/tools/dotc/config/Config.scala

```diff
@@ -49,6 +49,13 @@ object Config {
    */
   final val checkNoSkolemsInInfo = false
 
+  /** Check that Name#toString is not called directly from backend by analyzing
+   *  the stack trace of each toString call on names. This is very expensive,
+   *  so not suitable for continuous testing. But it can be used to find a problem
+   *  when running a specific test.
+   */
+  final val checkBackendNames = false
```

Comment:
I would propose to make this configurable as a flag. So that we can ask users to enable checked mode in case something looks very strange in their reports.